File: frontend/src/types/appointment.ts
import type { Suit } from "./suit";

export type CustomerRequest = {
  name: string;
  email: string;
  phone: string;
  message: string;
};

export type AppointmentRequest = {
  customer: CustomerRequest;
  suit?: Suit | null;
  price?: number | null;
};

export type AppointmentStatus =
  | "PENDING"
  | "CONTACTED"
  | "COMPLETED"
  | "CANCELLED";

export interface Appointment {
  id: number;
  customerName: string;
  email: string;
  phone: string;
  message: string;
  fabric?: string | null;
  lapel?: string | null;
  button?: string | null;
  price?: number | null;
  createdAt: string;
  status: AppointmentStatus;
}

File: backend/terzi-murat-backend/src/main/java/com/terzimurat/terzimuratbackend/entity/Appointment.java
package com.terzimurat.terzimuratbackend.entity;

import com.terzimurat.terzimuratbackend.entity.enums.AppointmentStatus;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "appointments")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Appointment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String customerName;

    @Column(nullable = false)
    private String email;

    @Column(nullable = false)
    private String phone;

    @Column(columnDefinition = "TEXT")
    private String message;

    @Column(nullable = false)
    private String fabric;

    @Column(nullable = false)
    private String lapel;

    @Column(nullable = false)
    private String button;

    @Column(nullable = false)
    private BigDecimal price;

    @CreationTimestamp
    @Column(updatable = false)
    private LocalDateTime createdAt;

    @PrePersist
    public void prePersist() {
        createdAt = LocalDateTime.now();
    }

    @Enumerated(EnumType.STRING)
    private AppointmentStatus status = AppointmentStatus.PENDING;
}